Adding Scala language (#100)
devstrons/heloworld: lang/scala.md

## Why to use Scala?

- Scala has an exact syntax, eliminating boilerplate code. Programs written in Scala require less code than similar programs written in Java.
- It is both an object-oriented language and a functional language. This combination makes Scala the right choice for web development.
- Scala uses an expressive typing system that ensures statistical abstraction is secure and consistent.
- It’s easy to learn, especially for programmers with an object-oriented background with Java or similar language.
- Scala is highly scalable, useful for building fault-tolerant, highly concurrent systems
- It’s ideal for data analytics when supported by tools like Apache Spark

## Setting up Scala development environment

- Step 1: Verifying Java Packages

> - The first thing we need to have is a Java Software Development Kit(SDK) installed on the computer. We need to verify these SDK packages and if not installed then install them. Open the command window and type in the following commands:

Reviewer: Provide a Wikipedia link to the Java SDK. Check other instances where it's needed as well.

- For users using Windows:
`C:\Users\Your_PC_username>java -version`

- Step 2: Installing Scala 
- We are done with installing the [Java SDK](https://en.wikipedia.org/wiki/Java_Development_Kit). Now let’s install the Scala packages. The best option to download these packages is to download from the official site only: https://www.scala-lang.org/download. 
The packages in the link above are approximately 100MB. Once the packages are downloaded, open the downloaded `.msi` file and install it.

- Step 3: Testing and running the Scala Commands
- Open the command prompt now and type in the following codes:
`C:\Users\Your_PC_username>scala`

## 💻 Popular Scala IDEs/Editors

- [IntelliJ IDEA](https://www.jetbrains.com/idea)
- [Vim](https://www.vim.org/download.php)
- [Sublime Text](https://www.sublimetext.com)
- [Atom Text editor](https://atom.io)
- [BlueJ](https://www.bluej.org)

## 😎 Some cool projects in Scala 

- [spark](https://github.com/apache/spark)
- [lila](https://github.com/lichess-org/lila)
- [SynapseML](https://github.com/microsoft/SynapseML)
- [The Algorithms - Scala](https://github.com/TheAlgorithms/Scala)

## 📑 Conclusion

> - Scala is a strong statically typed general-purpose programming language which supports both [object-oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming) and [functional programming](https://en.wikipedia.org/wiki/Functional_programming). Designed to be concise, many of Scala's design decisions are aimed to address criticisms of Java.

Reviewer: Not sure if we should use a list here (same for other parts of the document).

## Why to use Scala?

- Scala has an exact syntax, eliminating boilerplate code. Programs written in Scala require less code than similar programs written in Java.
- It is both an object-oriented language and a functional language. This combination makes Scala the right choice for web development.
- Scala uses an expressive typing system that ensures statistical abstraction is secure and consistent.
- It’s easy to learn, especially for programmers with an object-oriented background with Java or similar language.
- Scala is highly scalable, useful for building fault-tolerant, highly concurrent systems
- It’s ideal for data analytics when supported by tools like Apache Spark

## Setting up Scala development environment

- Step 1: Verifying Java Packages
  
  The first thing we need to have is a Java Software Development Kit(SDK) installed on the computer. We need to verify this SDK packages and if not installed then install them. Open the command window and type in the following commands:

- For Windows
```C:\Users\Your_PC_username>java -version```

> - Step 2: Now install Scala 

Reviewer: ```suggestion
- Step 2: Installing Scala
```

>   We are done with installing the java now let’s install the scala packages. The best option to download these packages is to download from the official site only: https://www.scala-lang.org/download/ 
>   The packages in the link above is the approximately of 100MB storage. Once the packages are downloaded then open the downloaded .msi file and install it.
> 
> - Step 3:Testing and Running the Scala Commands
>   
>   Open the command prompt now and type in the following codes
>   ```C:\Users\Your_PC_username>scala```
> 

Reviewer: ```suggestion
We are done with installing the Java SDK. Now let’s install the Scala packages. The best option to download these packages is to download from the official site only: https://www.scala-lang.org/download/ 
The packages in the link above are approximately 100MB. Once the packages are downloaded, open the downloaded `.msi` file and install it.

- Step 3: Testing and running the Scala Commands
  
Open the command prompt now and type in the following codes:

  `C:\Users\Your_PC_username>scala`
```

## Popular Scala IDEs/Editors

- IntelliJ IDEA
- Vim
- Sublime Text
- Atom Text editor

> - BlueJ

Reviewer: Could you please link to their official websites (if applicable)?
